src/js/components/MaskedInput/MaskedInput.js

```diff
@@ -72,7 +72,7 @@ const parseValue = (mask, value) => {
 
       maskIndex += 1;
       found = true;
-    } else if (item.options) {
+    } else if (item.options && item.restrictToOptions) {
```

Comment:
This looks like it will be a breaking change unless we test explicitly for `item.restrictToOptions !== false`.